lib/user-interface/react/src/components/chatbot/Message.tsx

```diff
@@ -17,47 +17,122 @@
+                            <Link onClick={() => {
+                                setSelectedImage(item);
+                                setSelectedMetadata(metadata);
+                                setShowImageViewer(true);
+                            }}>
+                                <img key={`${index}-Image`} src={item.image_url.url} alt='AI Generated' style={{ maxWidth:  '100%',  maxHeight: '30em', marginTop: '8px' }} />
+                            </Link>
+                            <ButtonDropdown
+                                items={[
+                                    { id: 'download-image', text: 'Download Image', iconName: 'download'},
+                                    { id: 'copy-image', text: 'Copy Image', iconName: 'copy'},
+                                    { id: 'regenerate', text: 'Regenerate Image(s)', iconName: 'refresh'}
+                                ]}
+                                ariaLabel='Control instance'
+                                variant='icon'
+                                onItemClick={async (e) => {
+                                    if (e.detail.id === 'download-image'){
+                                        const file = item.image_url.url.startsWith('https://') ?
+                                            await fetchImage(item.image_url.url)
+                                            : base64ToBlob(item.image_url.url.split(',')[1], 'image/png');
+                                        downloadFile(URL.createObjectURL(file), `${metadata?.imageGenerationParams?.prompt}.png`);
+                                    } else if (e.detail.id === 'copy-image') {
+                                        const copy = new ClipboardItem({ 'image/png':item.image_url.url.startsWith('https://') ?
+                                            await fetchImage(item.image_url.url) : base64ToBlob(item.image_url.url.split(',')[1], 'image/png') });
+                                        await navigator.clipboard.write([copy]);
+                                    } else if (e.detail.id === 'regenerate') {
+                                        setChatConfiguration({
+                                            ...chatConfiguration,
+                                            sessionConfiguration: {
+                                                ...chatConfiguration.sessionConfiguration,
+                                                imageGenerationArgs: {
+                                                    size: metadata?.imageGenerationParams?.size,
+                                                    numberOfImages: metadata?.imageGenerationParams?.n,
+                                                    quality: metadata?.imageGenerationParams?.quality
+                                                }
+                                            }
+                                        });
```

Comment:
We do this 3x places just in this file, and a bunch of other places as well. Is this something we can genericize, maybe using lodash merge or apply?